internal/util/debug/debug_test.go

```diff
@@ -0,0 +1,87 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package debug provides debug facilities.
+package debug
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/FerretDB/FerretDB/internal/util/must"
+	"github.com/FerretDB/FerretDB/internal/util/testutil"
+)
+
+func TestRunHandler(t *testing.T) {
+	t.Parallel()
+
+	// create and close TCP socket, to obtain a free port
+	l, err := net.ListenTCP("tcp", must.NotFail(net.ResolveTCPAddr("tcp", "localhost:0")))
+	require.NoError(t, err)
+
+	require.NoError(t, l.Close())
+
+	ctx, cancel := context.WithCancel(testutil.Ctx(t))
+
+	addr := l.Addr().(*net.TCPAddr)
+	started := make(chan struct{})
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	go func() {
+		RunHandler(ctx, addr.String(), prometheus.NewRegistry(), testutil.Logger(t), started)
```

Comment:
We most likely need to test `RunHandler` function as a whole instead of extracting `startedHandler`, because the handler relies on the `started` channel which is passed from the `RunHandler`.

Technically, we could still extract the handler and test it with `httptest`. In such scenario we need to refer to the channel created in test directly, but that leaves more space for some untracked bugs related to the `RunHandler` implementation